Bindings/Java/Matlab/Hopper_Device/OptimizeHopper.m

```diff
@@ -201,4 +182,57 @@
 
 end
 
+function plotSolution(x, problem)
+
+% Construct the actuator controls from solution 'x'. See 
+% CONSTRUCTCONTROLS above for details.
+[deviceControl, muscleExcitation, t] = constructControls(x, problem);
+                             
+% Plot optimized controls.
+figure;
+set(gcf, 'units', 'normalized', 'Position', [0.1 0.1 0.8 0.8])
+subplot(2,1,1);
+plot(deviceControl(1,:), deviceControl(2,:), 'b-', 'linewidth', 2)
+hold on
+plot(muscleExcitation(1,:), muscleExcitation(2,:), 'r-', 'linewidth', 2)
+ylim([-0.10 1.10]); xlim([0 5]); xticks(0:1:5);
+switch problem
+    case 'one_hop'
+        text_x = [x/2 t];
+        text_y = [1.05 -0.05];
+        labels = {'<--x(1)-->', 't1'};
+    case 'three_hops'
+        text_x = [(x([1 2 1 3 1])/2 + [0 t([1 2 3 4])]) t];
+        text_y = [1.05*ones(1,5) -0.05*ones(1,5)];
+        x_labels = arrayfun(@(x) sprintf('<--x(%i)-->', x), [1 2 1 3 1], ...
+            'UniformOutput', false);
```

Comment:
I did not know about `arrayfun()` but you should be able to get "iteration" using `sprintf()` on its own. I believe that `sprintf('%i ', [1, 2, 3])` will print `1 2 3` (it should repeat the string pattern).

Reply:
`sprintf()` will do the iteration, but returns a string. I need a cell array of strings for the `text()` argument, which `arrayfun(@func, 'UniformOutput', false)` returns. 

The other option to get a cell array would be `strsplit(sprintf('<--x(%i)-->,', [1 2 1 3 1]), ',')`. It gives an extra empty string in the last cell, but that could be ignored when creating the final label array:
`labels = [x_labels(1:end-1) t_labels(1:end-1)]`. This method would also avoid the line continuation that `arrayfun()` needs for the `UniformOutput` argument.